src/engine/Server.h

```diff
@@ -83,6 +84,14 @@ class Server {
 
   using SharedCancellationHandle = ad_utility::SharedCancellationHandle;
 
+  template <ad_utility::isInstantiation<absl::Cleanup> CancelTimeout>
+  struct CancellationHandleAndTimeoutTimerCancel {
+    SharedCancellationHandle handle_;
+    /// Function that when called cancels the timer that would otherwise
+    /// invoke the cancellation of the `handle_` via the time limit.
+    CancelTimeout cancelTimeout_;
+  };
```

Comment:
```suggestion
  };
  template <typename CancelTimeout>
  CancellationHandleAndTimoutTimerCancel(SharedCancellationHandle, CancelTimeout) -> CancellationHandleAndTimeoutTimerCancel<CancelTimeout>;
```
(Deduction guide, alternative to what you did, just a suggestion, not super important.
You can even repeat the concept here. But for one use only, leave it as is.